cot/src/response/into_response.rs

```diff
@@ -29,92 +27,176 @@ pub trait IntoResponse {
     ///
     /// # Errors
     /// Returns an error if the content type value is invalid.
-    fn with_content_type<V, VE>(self, content_type: V) -> cot::Result<Response>
+    fn with_content_type<V>(self, content_type: V) -> WithContentType<Self>
     where
-        V: TryInto<http::HeaderValue, Error = VE>,
-        VE: Into<http::Error>,
+        V: TryInto<http::HeaderValue>,
         Self: Sized,
     {
-        let content_type = content_type
-            .try_into()
-            .map_err(|e| ErrorRepr::from(e.into()))?;
-
-        self.into_response().map(|mut resp| {
-            resp.headers_mut()
-                .insert(http::header::CONTENT_TYPE, content_type);
-            resp
-        })
+        WithContentType {
+            inner: self,
+            content_type: content_type.try_into().ok(),
+        }
     }
 
     /// Modifies the response by setting the status code.
     ///
     /// # Errors
     /// Returns an error if the `IntoResponse` conversion fails.
-    fn with_status(self, status: StatusCode) -> cot::Result<Response>
+    fn with_status(self, status: StatusCode) -> WithStatus<Self>
     where
         Self: Sized,
     {
-        self.into_response().map(|mut resp| {
-            *resp.status_mut() = status;
-            resp
-        })
+        WithStatus {
+            inner: self,
+            status,
+        }
     }
 
     /// Modifies the response by setting the body.
     ///
     /// # Errors
     /// Returns an error if the `IntoResponse` conversion fails.
-    fn with_body(self, body: impl Into<Body>) -> cot::Result<Response>
+    fn with_body(self, body: impl Into<Body>) -> WithBody<Self>
     where
         Self: Sized,
     {
-        self.into_response().map(|mut resp| {
-            *resp.body_mut() = body.into();
-            resp
-        })
+        WithBody {
+            inner: self,
+            body: body.into(),
+        }
     }
 
     /// Modifies the response by inserting an extension.
     ///
     /// # Errors
     /// Returns an error if the `IntoResponse` conversion fails.
-    fn with_extension<T>(self, extension: T) -> cot::Result<Response>
+    fn with_extension<T>(self, extension: T) -> WithExtension<Self, T>
     where
         T: Clone + Send + Sync + 'static,
         Self: Sized,
     {
-        self.into_response().map(|mut resp| {
-            resp.extensions_mut().insert(extension);
+        WithExtension {
+            inner: self,
+            extension,
+        }
+    }
+}
+
+/// Returned by [`with_header`](IntoResponse::with_header) method.
+#[derive(Debug)]
+pub struct WithHeader<T> {
+    inner: T,
+    header: Option<(http::HeaderName, http::HeaderValue)>,
+}
+
+impl<T: IntoResponse> IntoResponse for WithHeader<T> {
+    fn into_response(self) -> cot::Result<Response> {
+        self.inner.into_response().map(|mut resp| {
+            if let Some((key, value)) = &self.header {
+                resp.headers_mut().append(key, value.clone());
+            }
```

Comment:
```suggestion
        self.inner.into_response().map(|mut resp| {
            if let Some((key, value)) = self.header {
                resp.headers_mut().append(key, value);
            }
```

Let's avoid this unnecessary clone!